benchmarks/benchmark_comparison.jl

```diff
@@ -0,0 +1,280 @@
+using Revise
+using Optim, Manopt
+using Manifolds
+using LineSearches
+using LinearAlgebra
+
+using Profile
+using ProfileView
+using BenchmarkTools
+using Plots
+using ManoptExamples
+using ImprovedHagerZhangLinesearch
+
+norm_inf(M::AbstractManifold, p, X) = norm(X, Inf)
+
+function f_rosenbrock(x)
+    result = 0.0
+    for i in 1:2:length(x)
+        result += (1.0 - x[i])^2 + 100.0 * (x[i + 1] - x[i]^2)^2
+    end
+    return result
+end
+function f_rosenbrock(::AbstractManifold, x)
+    return f_rosenbrock(x)
+end
+
+optimize(f_rosenbrock, [0.0, 0.0], Optim.NelderMead())
+
+function g_rosenbrock!(storage, x)
+    for i in 1:2:length(x)
+        storage[i] = -2.0 * (1.0 - x[i]) - 400.0 * (x[i + 1] - x[i]^2) * x[i]
+        storage[i + 1] = 200.0 * (x[i + 1] - x[i]^2)
+    end
+    return storage
+end
+
+optimize(f_rosenbrock, g_rosenbrock!, [0.0, 0.0], LBFGS())
+
+function g_rosenbrock!(M::AbstractManifold, storage, x)
+    g_rosenbrock!(storage, x)
+    if isnan(x[1])
+        error("nan")
+    end
+    riemannian_gradient!(M, storage, x, storage)
+    return storage
+end
+
+M = Euclidean(2)
+Manopt.NelderMead(M, f_rosenbrock)
+
+qn_opts = quasi_Newton(
+    M,
+    f_rosenbrock,
+    g_rosenbrock!,
+    [0.0, 0.0];
+    evaluation=InplaceEvaluation(),
+    return_state=true,
+)
+
+function test_f(f, g!, x0, N::Int)
+    M = Euclidean(N)
+    return quasi_Newton(M, f, g!, x0; evaluation=InplaceEvaluation(), return_state=true)
+end
+
+function prof()
+    N = 32
+    x0 = zeros(N)
+    test_f(f_rosenbrock, g_rosenbrock!, x0, N)
+
+    Profile.clear()
+    @profile for i in 1:100000
+        test_f(f_rosenbrock, g_rosenbrock!, x0, N)
+    end
+    return ProfileView.view()
+end
+
+to_optim_manifold(::Manifolds.Sphere) = Optim.Sphere()
+to_optim_manifold(::Euclidean) = Optim.Flat()
+
+abstract type AbstractOptimConfig end
+struct ManoptQN <: AbstractOptimConfig end
+
+function benchmark_time_state(
+    ::ManoptQN,
+    M::AbstractManifold,
+    N,
+    f,
+    g!,
+    x0,
+    stepsize::Manopt.Stepsize,
+    mem_len::Int,
+    gtol::Real;
+    kwargs...,
+)
+    manopt_sc = StopWhenGradientNormLess(gtol; norm=norm_inf) | StopAfterIteration(1000)
+    mem_len = min(mem_len, manifold_dimension(M))
+    bench_manopt = @benchmark quasi_Newton(
+        $M,
+        $f,
+        $g!,
+        $x0;
+        stepsize=$(stepsize),
+        evaluation=$(InplaceEvaluation()),
+        memory_size=$mem_len,
+        stopping_criterion=$(manopt_sc),
+        debug=[],
+        $kwargs...,
+    )
+    manopt_state = quasi_Newton(
```

Comment:
Would it maybe make sense to do this call before the benchmark (Line 97)?

Reply:
It shouldn't really matter, benchmarking macro already takes care of compilation time.